packages/stark-core/src/modules/http/helpers/http-helpers.ts

```diff
@@ -0,0 +1,45 @@
+import { HttpParameterCodec, HttpParams } from "@angular/common/http";
+import { StarkQueryParam } from "@nationalbankbelgium/stark-core";
+import { CustomEncoder } from "../entities/custom-encoder";
+
+/**
+ *  @ignore
+ */
+const _reduce: Function = require("lodash/reduce");
+
+/**
+ * A custom implementation of HttpParameterCodec to correctly encode query parameters.
+ * @link https://github.com/NationalBankBelgium/stark/issues/1130
+ */
+export const CUSTOM_ENCODER: HttpParameterCodec = new CustomEncoder();
+
+/**
+ * Converts the Map<string, StarkQueryParam> required by the service into a HttpParams object required by Angular
+ * @param starkQueryParam - params to convert
+ */
+export function convertStarkQueryParamsIntoHttpParams(starkQueryParam: Map<string, StarkQueryParam>): HttpParams {
+	return _reduce(
+		convertMapIntoObject(starkQueryParam), // convert to object
+		(httpParams: HttpParams, starkParams: StarkQueryParam, key: string) =>
+			starkParams
+				? Array.isArray(starkParams)
+					? _reduce(starkParams, (acc: HttpParams, entry: string) => acc.append(key, entry), httpParams)
+					: httpParams.set(key, starkParams)
+				: httpParams.set(key, ""),
```

Comment:
Why is this key set to ""?

Reply:
set is defined as `set(param: string, value: string): HttpParams;` so when it's `undefined` I set it to an empty string